core/solver/multigrid.cpp

```diff
@@ -762,6 +762,11 @@ void Multigrid::generate()
     GKO_ASSERT_EQ(level > 0, true);
     auto last_mg_level = mg_level_list_.back();
 
+    using ws = workspace_traits<Multigrid>;
+    this->setup_workspace();
+    this->create_state();
+    cache_.state->generate(this->get_system_matrix().get(), this, 1);
```

Comment:
It moves the initialization from first apply to generation.
Do I miss anything here?

Reply:
Yes, it is mostly to remove any initalizations from apply